app/controllers/spotlight/resources/upload_controller.rb

```diff
@@ -0,0 +1,36 @@
+module Spotlight::Resources
+  class UploadController < ApplicationController
+    before_filter :authenticate_user!
+
+    load_resource :exhibit, class: Spotlight::Exhibit
+    before_filter :build_resource, only: [:create]
+
+    load_and_authorize_resource class: 'Spotlight::Resources::Upload', through_association: "exhibit.resources", instance_name: 'resource'
+
+    def create
+      @resource.attributes = resource_params
+
+      if @resource.save
+        flash[:notice] = t('spotlight.resources.upload.success')
+        if params["add-and-continue"]
+          redirect_to new_upload_exhibit_catalog_index_path(@resource.exhibit)
+        else
+          redirect_to admin_exhibit_catalog_index_path(@resource.exhibit)
+        end
+      else
+        flash[:error] = t('spotlight.resources.upload.error')
+        redirect_to admin_exhibit_catalog_index_path(@resorce.exhibit)
+      end
+    end
+
+    private
+    def build_resource
```

Comment:
This may be leftover from when trying to subclass the resources controller.  I'll check to see if cancan is doing this for us.